### PR TITLE
Finished all views for Home Component

### LINKED ISSUE #14: Build out Home component and it's child components
## Description

1. `<LeftSidebar />`
2. `<MainContent />`
3.  `<RightSidebar />`

#### Component Representation

homediagram

#### Trello Reference

hometrello

### PR DESCRIPTION
**Pull request by nickeyvee**
### Description

- Refactored many of our styles and placed them into **StyledComponents**  for re-use throughout our project.
- Created **PopOver.js**. This is a [high order component](https://reactjs.org/docs/higher-order-components.html) that takes a two components as arguments, a button and content. Now you can render custom dialog's everywhere.
- Repeated **PopOver** dialogs for creating teams, creating boards, and deleting teams. New components include **CreateTeam.js, CreateBoard.js,** and **DeleteTeam.js** These are passed to **PopOver.js** to be rendered as a dialog.
- The new **TeamViewer.js** component allows us an interface to customize are teams in multiple ways such as adding a description, adding and removing boards for our team, and managing our members.
- refactored **index.js** file for each of our wrapper (parent) components.

Keep in mind these are only changes to the frontend code.

closes #14 

screenshot 185

screenshot 186

screenshot 188

🔥 Thank you!

### CODE REVIEW

File: src/components/CreateBoardButton/CreateBoardButton.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  setButtonRef = node => {
+    this.buttonRef = node
+  }
+
+  render() {
+    const { togglePopOver, buttonType } = this.props
+    return (
+      <div ref={this.setButtonRef}>
+        {
+          this.props && buttonType && buttonType === 'card' ?
```
Comment (albertoescala) [inline src/components/CreateBoardButton/CreateBoardButton.js]: @nickeyvee why do you check if `this.props` exists? 🤔

File: src/components/CreateTeam/CreateTeam.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
+import React from 'react'
+
+import { Button, Heading, Icon, Input, TextArea, Wrapper } from '../StyledComponents'
+
+const CreateTeam = props => {
+  return (
+    <Wrapper width="100%">
+      <Heading>
+        {props && props.hasChevron ?
```
Comment (albertoescala) [inline src/components/CreateTeam/CreateTeam.js]: Same as above.

File: src/components/Home/components/BoardsHome/BoardsHome.js [hunk 3]
```diff
@@ -33,12 +34,13 @@ const BoardsQuery = gql`
...
 
 const BoardsHome = props => {
+  const CreateBoardPopOver = PopOver(CreateBoard, CreateBoardButton)
   return (
     <div>
       <Query query={BoardsQuery} variables={{ id: props.userId }}>
         {({ loading, data }) => {
           return (
-            !loading && data && (
+            !loading && data && data.User && data.User.boards && (
```
Comment (albertoescala) [inline src/components/Home/components/BoardsHome/BoardsHome.js]: I think that check `data.User.boards` is not necessary.
- Have you try only with `data.User`?

`!loading && data.User && (...)`

File: src/components/Home/components/LeftSidebar/LeftSidebar.js [hunk 4]
```diff
@@ -53,8 +46,10 @@ const LeftSidebar = props => {
           return (
             !loading && data && data.User && data.User.teams.map(team => (
```
Comment (albertoescala) [inline src/components/Home/components/LeftSidebar/LeftSidebar.js]: Same as above.

File: src/components/Home/components/TeamViewer/components/EditTeam/EditTeam.js [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+  componentWillReceiveProps = nextProps => {
+    const {
+      id,
+      name,
+      desc,
+      website
+    } = nextProps.team
+    this.setState({
+      teamId:
+        id,
```
Comment (albertoescala) [inline src/components/Home/components/TeamViewer/components/EditTeam/EditTeam.js]: Lol, what happened here? 😆 
```
this.setState({
  teamId: id,
  .
  .
  .
```

**Review by albertoescala — CHANGES_REQUESTED** [review]

File: src/components/Toolbar/components/Avatar/Avatar.js [hunk 6]
```diff
@@ -0,0 +1,12 @@
+import React from 'react'
+import { Button, Img } from './Avatar.styles'
+
+const Avatar = props => {
+  return (
+    <Button avatar onClick={() => props.togglePopOver()}>
+      <Img alt="" src={localStorage.getItem('picture')} />
```
Comment (albertoescala) [inline src/components/Toolbar/components/Avatar/Avatar.js:7]: `alt=""` 🤣 Nice hack!

**Review by albertoescala — COMMENTED** [review]
💬

File: src/components/Home/components/BoardsHome/BoardsHome.js (hunk 3 above)
Comment (nickeyvee) [reply src/components/Home/components/BoardsHome/BoardsHome.js]: I was a little paranoid with this. I will go ahead and refactor these.

Comment (albertoescala) [reply src/components/Home/components/BoardsHome/BoardsHome.js]: @nickeyvee That's ok. Try it.

**Review by albertoescala — APPROVED** [review]

**Pull request merged**